ghcide/src/Development/IDE/Spans/Documentation.hs

```diff
@@ -68,11 +69,14 @@ getDocumentationTryGhc env mod n = head <$> getDocumentationsTryGhc env mod [n]
 
 getDocumentationsTryGhc :: HscEnv -> Module -> [Name] -> IO [SpanDoc]
 getDocumentationsTryGhc env mod names = do
-  res <- catchSrcErrors (hsc_dflags env) "docs" $ (fmap . fmap) snd $ fmap Map.toList $ getDocsBatch env mod names
+  res <- fun
   case res of
       Left _    -> return []
-      Right res -> zipWithM unwrap res names
+      Right res -> zipWithM unwrap (fmap snd res) names
   where
+    fun :: IO (Either [FileDiagnostic] [(Name, Either String (Maybe HsDocString, Map.Map Int HsDocString))])
+    fun = catchSrcErrors (hsc_dflags env) "docs" $ Map.toList <$> getDocsBatch env mod names
+
     unwrap (Right (Just docs, _)) n = SpanDocString docs <$> getUris n
```

Comment:
Now, in the next commits I start to passing-through the `Map` retrieved from `getDocsBatch` further & further to reach the `getDocumentationsTryGhc` API point